unic/char/range/src/range.rs

```diff
@@ -148,6 +149,25 @@ impl CharRange {
         self.low <= ch && ch <= self.high
     }
 
+    /// Determine the ordering of this range and a character.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the range is empty. This fn may be adjusted in the future to not panic
+    /// in optimized builds. Even if so, an empty range will never compare as `Ordering::Equal`.
+    ///
+    pub fn cmp(&self, ch: char) -> Ordering {
+        // possible optimization: only assert this in debug builds
+        assert!(!self.is_empty(), "Cannot compare empty range's ordering");
```

Comment:
Awesome! We should consider this, but as long as we make sure the `release` mode behavior can't make more troubles, like resulting in an infinite loop if called on empty range.

Another possible solution would be to consider having separate concrete implementations for empty and non-empty ranges. (I think `std` had something like that.) The rational would be that we have much less range manipulation than we have lookups like this, therefore knowing in advance if the range is empty or not can save lots of unnecessary checks here and there.

What do you think?